course/views.py

```diff
@@ -35,10 +35,11 @@ def get_common_objects(self):
         self.external_user_label = settings_text('EXTERNAL_USER_LABEL')
         self.instances = []
         prio2 = []
-        treshold = timezone.now() - datetime.timedelta(days=10)
+        start_treshold = timezone.now() - datetime.timedelta(days=10)
+        end_threshold = timezone.now() - datetime.timedelta(days=30)
         for instance in CourseInstance.objects.get_visible(self.request.user)\
-                .filter(ending_time__gte=timezone.now()):
-            if instance.starting_time > treshold:
+                .filter(ending_time__gte=end_threshold):
+            if instance.starting_time > start_treshold:
                 self.instances += [instance]
```

Comment:
This is a good place to show how slow python method resolution is too:

```
>>> timeit('for i in range(50): a.append(i)', 'a = []', number=1000000)
1.7041928516700864
>>> timeit('for i in range(50): _(i)', 'a = []; _ = a.append', number=1000000)
1.206302935257554
``

In this test, doing the attribute resolution adds 40% overhead, hence when same attribute is resolved _multi_ times, e.g., in a loop, caching that attribute resolution might be beneficial. Of course, this makes code often less readable, thus one should consider if the performance benefit is more or less imporant than the readability. I just think this is important fact to know about Python.

Reply:
That's a good point and good to know. Actually I had not myself seen the `+= [instance]` used before, it was readily done in the previous code. I could change those to append, it is no problem.

Reply:
I tried with list comprehension but it looked quite messy, like you said it might, so I went with append. In my opinion it looks now clean, the code is readable and it works well in this case.

Reply:
>  the code is readable

This is the most important aspect. A code which is fast, but unmaintainable, would still cost more, because the maintenance is one of the biggest costs for a program.